node_modules/gh-series/tests/test-patterns.js

```diff
@@ -0,0 +1,259 @@
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+var _ = require('lodash');
+var assert = require('assert');
+var moment = require('moment');
+
+var EventsTestUtil = require('gh-events/tests/util');
+var SeriesTestUtil = require('./util');
+
+var FullPattern = require('gh-series/lib/internal/patterns/cambridge/fullpattern').FullPattern;
+
+describe('Series', function() {
+
+    describe('Aggregation', function() {
+
+        /**
+         * Create a serie with 3 events in it. The events will aggregate both in time, place and organisers
+         *
+         * @param  {Function}       callback            Standard callback function
+         * @param  {Event}          callback.eventA     The first event
+         * @param  {Event}          callback.eventB     The second event
+         * @param  {Event}          callback.eventC     The thid event
+         * @param  {Serie}          callback.serie      The serie that holds the events
+         */
+        var setupAggregation = function(callback) {
+            // Create 3 events that hold aggregatable information. Keep in mind
+            // that we have to provide the start/end times in UTC
+            EventsTestUtil.assertCreateEvent(global.tests.admins.cam2014.client, 'A', '2014-10-28T13:00:00', '2014-10-28T14:00:00', {'organiserOther': ['Prof William Lawsons'], 'location': 'Room 1'}, function(eventA) {
+                EventsTestUtil.assertCreateEvent(global.tests.admins.cam2014.client, 'A', '2014-11-04T13:00:00', '2014-11-04T14:00:00', {'organiserOther': ['Dr Jack Daniels'], 'location': 'Room 1'}, function(eventB) {
+                    EventsTestUtil.assertCreateEvent(global.tests.admins.cam2014.client, 'A', '2014-11-11T13:00:00', '2014-11-11T14:00:00', {'organiserOther': ['Prof William Lawsons'], 'location': 'Room 2'}, function(eventC) {
+
+                        // Create a serie and add the events
+                        SeriesTestUtil.assertCreateSerie(global.tests.admins.cam2014.client, 'Test serie', {}, function(serie) {
+                            SeriesTestUtil.assertAddSeriesEvents(global.tests.admins.cam2014.client, serie.id, [eventA.id, eventB.id, eventC.id], function() {
+
+                                // Get the serie and check if the aggregated metadata is correct
+                                SeriesTestUtil.assertGetSerie(global.tests.admins.cam2014.client, serie.id, null, function(serie) {
```

Comment:
Have we confirmed that generating the summary asynchronously is going to work for the admin UI?